paasta_tools/paasta_cluster_boost.py

```diff
@@ -0,0 +1,149 @@
+from paasta_tools.utils import paasta_print
+
+log = logging.getLogger(__name__)
+
+
+def parse_args():
+    """Parses the command line arguments passed to this script"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-r', '--region',
+        type=str,
+        required=True,
+        help="name of the region where the pool is. eg: useast1-prod",
+    )
+    parser.add_argument(
+        '-p', '--pool',
+        type=str,
+        default='default',
+        help="Name of the pool you want to increase the capacity. Default is 'default' pool.",
+    )
+    parser.add_argument(
+        '-b', '--boost',
+        type=float,
+        default=cluster_boost.DEFAULT_BOOST_FACTOR,
+        help="Boost factor to apply. Default is 1.5. A big failover should be 2, 3 is the max.",
+    )
+    parser.add_argument(
+        '-d', '--duration',
+        type=int,
+        default=cluster_boost.DEFAULT_BOOST_DURATION,
+        help="Duration of the capacity boost in minutes. Default is 40min.",
+    )
+    parser.add_argument(
+        '-f', '--force',
+        action='store_true',
+        dest='override',
+        help="Replace an existing boost. Default is false",
+    )
+    parser.add_argument(
+        'action',
+        choices=[
+            'set',
+            'status',
+            'clear',
+        ],
+        help="You can set, get or clear a boost.",
+    )
+    parser.add_argument(
+        '-v', '--verbose',
+        action='count',
+        dest="verbose",
+        default=0,
+        help="Print out more output.",
+    )
+    return parser.parse_args()
+
+
+def check_pool_exist(pool: str, region: str) -> bool:
+    """ Check that we have slaves in the provided pool and region
+    """
+    system_paasta_config = load_system_paasta_config()
+    expected_slave_attributes = system_paasta_config.get_expected_slave_attributes()
+    if expected_slave_attributes is None:
+        return False
+
+    region_pool_pairs = []
+    for slave in expected_slave_attributes:
+        slave_pool = slave['pool']
+        slave_region = slave['region']
+        region_pool_pair = (slave_region, slave_pool)
+        if region_pool_pair not in region_pool_pairs:
+            region_pool_pairs.append(region_pool_pair)
+
+    return (region, pool) in region_pool_pairs
+
+
+def paasta_cluster_boost():
+    """ Set, Get or clear a boost on a paasta cluster for a given pool in a given region
+    :returns: None
+    """
+    args = parse_args()
+
+    if args.verbose >= 2:
+        logging.basicConfig(level=logging.DEBUG)
+    elif args.verbose == 1:
+        logging.basicConfig(level=logging.INFO)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+
+    action = args.action
+    region = args.region
+    pool = args.pool
+
+    if action == 'set':
+        # Let's disallow people to set boost on a non existing pool
+        if not check_pool_exist(pool=pool, region=region):
```

Comment:
Is there a reason this is only checked on 'set'?

Reply:
Initially, because I was using mesos state, making a call to this function very expensive but @solarkennedy pointed me to system_paasta_config.get_expected_slave_attributes() which is  less expensive so I guess we could sysmatically check that.